pythonforandroid/recipes/jpeg/__init__.py

```diff
@@ -1,35 +1,78 @@
-from pythonforandroid.recipe import NDKRecipe
+from pythonforandroid.recipe import Recipe
 from pythonforandroid.logger import shprint
 from pythonforandroid.util import current_directory
 from os.path import join, exists
+from os import environ, uname
+from glob import glob
 import sh
 
 
-class JpegRecipe(NDKRecipe):
+class JpegRecipe(Recipe):
+    '''
+    .. versionchanged:: 0.6.0
+        rewrote recipe to be build with clang and updated libraries to latest
+        version of the official git repo.
+    '''
     name = 'jpeg'
-    version = 'linaro-android'
-    url = 'git+https://git.linaro.org/people/tomgall/libjpeg-turbo/libjpeg-turbo'
+    version = '2.0.1'
+    url = 'https://github.com/libjpeg-turbo/libjpeg-turbo/archive/{version}.tar.gz'  # noqa
+    # we will require this below patch to build the shared library
+    # patches = ['remove-version.patch']
 
-    patches = ['build-static.patch']
+    def should_build(self, arch):
+        return not exists(join(self.get_build_dir(arch.arch),
+                               'libturbojpeg.a'))
 
-    generated_libraries = ['libjpeg.a']
+    def build_arch(self, arch):
+        super(JpegRecipe, self).build_arch(arch)
+        build_dir = self.get_build_dir(arch.arch)
 
-    def prebuild_arch(self, arch):
-        super(JpegRecipe, self).prebuild_arch(arch)
+        # TODO: Fix simd/neon
+        with current_directory(build_dir):
+            env = self.get_recipe_env(arch)
+            toolchain_file = join(self.ctx.ndk_dir,
+                                  'build/cmake/android.toolchain.cmake')
 
-        build_dir = self.get_build_dir(arch.arch)
-        app_mk = join(build_dir, 'Application.mk')
-        if not exists(app_mk):
-            shprint(sh.cp, join(self.get_recipe_dir(), 'Application.mk'), app_mk)
-        jni_ln = join(build_dir, 'jni')
-        if not exists(jni_ln):
-            shprint(sh.ln, '-s', build_dir, jni_ln)
+            shprint(sh.rm, '-f', 'CMakeCache.txt', 'CMakeFiles/')
+            shprint(sh.cmake, '-G', 'Unix Makefiles',
+                    '-DCMAKE_SYSTEM_NAME=Android',
+                    '-DCMAKE_SYSTEM_PROCESSOR={cpu}'.format(cpu='arm'),
+                    '-DCMAKE_POSITION_INDEPENDENT_CODE=1',
+                    '-DCMAKE_ANDROID_ARCH_ABI={arch}'.format(arch=arch.arch),
+                    '-DCMAKE_ANDROID_NDK=' + self.ctx.ndk_dir,
```

Comment:
Minor, why don't you use the same format string  syntax as before/after for consistency?